### PR TITLE
Add multi-GPU support for Tensorflow multi-stream with VGPUs

### PR DESCRIPTION
**Pull request by xprotobeast2**

### CODE REVIEW

File: tools/patch/tensorflow/cc/saved_model/loader.cc [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
-+  const std::string& vdl =
-+    lsession_options.config.gpu_options().visible_device_list();
-+  if (!vdl.empty() && (vdl[0] == '/')) {
-+    graph::SetDefaultDevice(vdl, bundle->meta_graph_def.mutable_graph_def());
-+    lsession_options.config.mutable_gpu_options()->clear_visible_device_list();
++  const std::string& alloc_type =
++    lsession_options.config.gpu_options().allocator_type();
++  if (!alloc_type.empty() && (alloc_type[0] == '/')) {
++    graph::SetDefaultDevice(alloc_type, bundle->meta_graph_def.mutable_graph_def());
++    lsession_options.config.mutable_gpu_options()->clear_allocator_type();
```
Comment (GuanLuo) [inline tools/patch/tensorflow/cc/saved_model/loader.cc]: @tanmayv25 Is this related to what you are trying to do with TF-TRT models?

Comment (xprotobeast2) [reply tools/patch/tensorflow/cc/saved_model/loader.cc]: Tanmay seems to also be working on this code. This may lead to some merge issues later.

Comment (GuanLuo) [reply tools/patch/tensorflow/cc/saved_model/loader.cc]: It is fine as long as you guys are working on different things. I just worried that he was working on something that has been implemented.

File: src/core/model_repository_manager.cc [hunk 2]
```diff
@@ -86,13 +90,30 @@ BuildBackendConfigMap(
...
+#ifdef TRTIS_ENABLE_GPU
+    int device_cnt = 0;
+    cudaError_t cuerr = cudaGetDeviceCount(&device_cnt);
+    if (cuerr == cudaErrorNoDevice) {
+      device_cnt = 0;
+    } else if (cuerr != cudaSuccess) {
+      LOG_ERROR << "unable to get number of CUDA devices while building "
+                   "BackendConfigMap: "
+                << cudaGetErrorString(cuerr);
+    }
```
Comment (deadeyegoodwin) [inline src/core/model_repository_manager.cc]: set device_cnt = 0 since not sure what cudaGetDeviceCount did to it in error case

**Pull request merged**